src/middlewared/middlewared/plugins/pool.py

```diff
@@ -1201,21 +1201,40 @@ async def remove(self, oid, options):
 
         await self.middleware.call('zfs.pool.remove', pool['name'], found[1]['guid'])
 
+        if found[1]['type'] != 'DISK':
+            disk_paths = [d['path'] for d in found[1]['children']]
+        else:
+            disk_paths = [found[1]['path']]
+
         if osc.IS_FREEBSD:
             await self.middleware.call('pool.sync_encrypted', oid)
 
-            if found[1]['path'].endswith('.eli'):
-                devname = found[1]['path'].replace('/dev/', '')[:-4]
-                await self.middleware.call('disk.geli_detach_single', devname)
+            for disk_path in disk_paths:
+                if disk_path.endswith('.eli'):
+                    devname = disk_path.replace('/dev/', '')[:-4]
+                    await self.middleware.call('disk.geli_detach_single', devname)
 
-        disk = await self.middleware.call(
-            'disk.label_to_disk', found[1]['path'].replace('/dev/', '')
-        )
-        if disk:
-            wipe_job = await self.middleware.call('disk.wipe', disk, 'QUICK')
+        wipe_jobs = []
+        for disk_path in disk_paths:
+            disk = await self.middleware.call(
+                'disk.label_to_disk', disk_path.replace('/dev/', '')
+            )
+            if disk:
+                wipe_job = await self.middleware.call('disk.wipe', disk, 'QUICK')
+                wipe_jobs.append((disk, wipe_job))
+
+        failed_wipes = []
+        for disk, wipe_job in wipe_jobs:
             await wipe_job.wait()
             if wipe_job.error:
-                raise CallError(f'Failed to wipe disk {disk}: {wipe_job.error}')
+                failed_wipes.append((disk, wipe_job.error or ''))
```

Comment:
You could probably form the error_str here? Not sure there is any benefit in looping over the failures again?

Reply:
I agree :)